---
title: "Data Science Tapas"
subtitle: "Genome Wide Association Studies with Hail Scalable Software"
author: "Heidi Steiner"
date: "March 16, 2023"
footer: "https://github.com/ua-data7/DS-Tapas/Workshops/Spring2023/2023-Mar-16"
format:
  revealjs:
    theme: [default, assets/custom.scss]
    logo: assets/dsi.png
    auto-play-media: true
---

## On Today's Menu

-   Installations
-   Discuss
    -   Hail + scalability üèóÔ∏è

    -   Genome wide association studies (GWAS) üß¨
-   Hands-on
    -   perform GWAS with hail üå®Ô∏è

## Prepare Workspace

-   Load CyVerse JupyterLab DataScience App

-   Install Java

<!-- -->

    sudo apt update
    sudo apt install openjdk-8-jdk

-   Install hail

<!-- -->

    pip install hail

::: notes
double check that you need argparse and numpy here
:::

## What is Hail?

![](assets/images/image-2094204328.png){fig-align="center" width="274"}

-   Open source data science library

-   Scale-able genomic software

-   Unified genomic data representation

-   Community ü§ó

::: notes
slice, dice, query, and model data of any kind

easy to use with small and biobank genomic datasets

matrix table is the same for all types of genomic data

forum + chatroom
:::

## The scalability problem

It's not feasible to process tens or hundreds of thousands of whole genomes on a single computer

Users can worry about the contents of their pipelines, rather than how to parallelize them

## Genome wide association studies

::: columns
::: {.column width="50%"}
Statistical method to survey large amounts of genetic variants for a relationship with a particular disease (or another trait)
:::

::: {.column width="50%"}
![](assets/images/gwas.png){fig-align="center"}
:::
:::

## Let's try it ‚ñ∂Ô∏è

Back to CyVerse

## Buen Provecho!

-   [Provide Tapas Feedback](https://uarizona.co1.qualtrics.com/jfe/form/SV_dmxCZsorVVr5lA2)
-   [Request a consulatation](https://uarizona.co1.qualtrics.com/jfe/form/SV_7V8weDrKwCMbXCK)
-   Mastering hail takes practices, find more [hail resources online](https://hail.is/docs/0.2/index.html)!
